nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,561 @@
+    statistical_maps = make_statistical_maps(model, contrasts)
+    html_design_matrices = _make_html_for_design_matrices(model)
+    roi_plot_html_code = _make_roi_plot(roi_img, bg_img)
+    all_components = _make_report_components(
+            statistical_maps=statistical_maps,
+            contrasts_plots=contrast_plots,
+            threshold=threshold,
+            alpha=alpha,
+            cluster_threshold=cluster_threshold,
+            height_control=height_control,
+            min_distance=min_distance,
+            bg_img=bg_img,
+            display_mode=display_mode,
+            plot_type=plot_type,
+            )
+    all_components_text = '\n'.join(all_components)
+    report_values = {'page_title': page_title,
+                     'page_heading_1': page_heading_1,
+                     'page_heading_2': page_heading_2,
+                     'model_attributes': model_attributes_html,
+                     'all_contrasts_with_plots': all_contrast_plots_text,
+                     'design_matrices': html_design_matrices,
+                     'roi_plot': roi_plot_html_code,
+                     'component': all_components_text,
+                     }
+    report_text = report_template.safe_substitute(**report_values)
+    report = HTMLDocument(report_text)
+    report.width = nb_width  # for better visual experience in Jupyter Notebooks.
+    report.height = nb_height
+    return report
+
+
+def _make_contrasts_dict(contrasts):
+    """ Accepts contrasts and returns a dict of them.
+    with the names of contrasts as keys.
+    
+    If contrasts is:
+      dict then returns it unchanged.
+      
+      string or list/tuple of strings, returns a dict where key==values
+    
+    Parameters
+    ----------
+    contrasts: str, List/Tuple[str], Dict[str, str or np.array]
+        Contrast information
+    
+    Returns
+    -------
+    contrasts: Dict[str, np.array or str]
+        Contrast information, as a dict
+    """
+    contrasts = [contrasts] if isinstance(contrasts, str) else contrasts
+    if not isinstance(contrasts, dict):
+        contrasts = {contrast_: contrast_ for contrast_ in contrasts}
+    return contrasts
+
+
+def _make_dict_of_contrast_plots(contrasts, model):
+    """
+    Accepts dict of contrasts and First or Second Level Model
+    with fitted design matrices and generates
+    a dict of contrast name & svg code for corresponding contrast plot.
+    
+    Parameters
+    ----------
+    contrasts: Dict[str, np.array or str]
+        Contrast information, as a dict
+    
+    model: FirstLevelModel or SecondLevelModel object
+        FLM or SLM with fitted design matrices.
+
+    Returns
+    -------
+    contrast_plots: Dict[str, svg img]
+        Dict of contrast name and svg code for corresponding contrast plot.
+    """
+    contrast_plots = {}
+    for design_matrix in model.design_matrices_:
+        for contrast_name, contrast_data in contrasts.items():
```

Comment:
what if a user wants to provide a list of contrasts (one for each design matrix), as can be done for `FirstLevelModel.compute_contrast`?

Reply:
The model already is supposed to have the info for the design matrix. The user first generates the model using standard `FirstLevelModel` or `SecondLevelModel` object & then pass the completed model object into the reporter.

Reply:
Currently, there is no way of specifying a list of first level contrasts to be passed to a second level model. I have made that clearer in the docstring. You can discuss with BT & i can implement the consensus.

Reply:
I've never had this request so far...